src/Main.elm

```diff
@@ -1,10 +1,418 @@
+                        nextLocation =
+                            Store.getFolderByID folderID store
+                                |> Maybe.andThen .parent
+                                |> Maybe.map Folder
+                                |> Maybe.withDefault Root
+                    in
+                    Succeed store { state | location = nextLocation }
+            , Cmd.none
+            )
+
+        ( GoUp, _ ) ->
+            ( model, Cmd.none )
+
+        ( SetFilterBy nextFilter, Succeed store state ) ->
+            ( Succeed store
+                { state
+                    | location = Root
+                    , filterBy = nextFilter
+                }
+            , Cmd.none
+            )
+
+        ( SetFilterBy _, _ ) ->
+            ( model, Cmd.none )
+
+
+
+-------------
+-- V I E W --
+-------------
+
+
+stylesButton : List (Element.Attribute msg)
+stylesButton =
+    [ Element.width Element.fill
+    , Element.padding 20
+    ]
+
+
+viewFolder : AudienceFolder -> Element Msg
+viewFolder folder =
+    button
+        (Element.Background.color (Element.rgb255 26 113 153)
+            :: Element.Font.color (Element.rgb255 255 255 255)
+            :: stylesButton
+        )
+        { onPress = Just (GoDown folder.id)
+        , label = paragraph [] [ text ("→ " ++ folder.name) ]
+        }
+
+
+viewAudience : Audience -> Element msg
+viewAudience audience =
+    paragraph
+        (Element.Background.color (Element.rgb255 24 151 205)
+            :: Element.Font.color (Element.rgb255 255 255 255)
+            :: stylesButton
+        )
+        [ text audience.name ]
+
+
+viewGoUp : AudienceFolder -> Element Msg
+viewGoUp parentFolder =
+    button
+        (Element.Border.solid
+            :: Element.Border.width 2
+            :: Element.Border.color (Element.rgb255 26 113 153)
+            :: Element.Font.color (Element.rgb255 26 113 153)
+            :: stylesButton
+        )
+        { onPress = Just GoUp
+        , label = paragraph [] [ text ("↑ " ++ parentFolder.name) ]
+        }
+
+
+viewLevelColumn : List (Element msg) -> Element msg
+viewLevelColumn =
+    column
+        [ Element.width Element.fill
+        , Element.height Element.fill
+        , Element.scrollbarY
+        , Element.spacing 8
+        ]
+
+
+viewLevel : Element Msg -> AudienceLevel -> Element Msg
+viewLevel toUp level =
+    viewLevelColumn
+        (toUp :: List.map viewFolder level.folders ++ List.map viewAudience level.audiences)
+
+
+view404 : AudienceFolderID -> Element msg
+view404 folderID =
+    paragraph
+        []
+        [ text ("The folder `" ++ String.fromInt folderID ++ "` doesn't exist")
+        ]
+
+
+viewFilterButton : List (Element.Attribute msg) -> Element msg -> Element msg
+viewFilterButton attributes label =
+    button
+        (Element.width Element.fill
+            :: Element.paddingXY 6 18
+            :: Element.Font.center
+            :: attributes
+        )
+        { onPress = Nothing
```

Comment:
Are there some bugs or problems when using `onPress`? I see you're passing `onClick` in `attributes` instead - but I'd expect the events to be passed into `onPress`...

Reply:
Yes, that's right. I find a bit strange the way of `button` arguments like
```elm
{ onPress : Maybe msg
}
```

Reply:
I don't think there's a bug in elm-ui. onPress works fine for me. Maybe it was confusing why onPress message is wrapped in Maybe? AFAIK that was a design decision of the author with Maybe representing disabled button (to express that onPress on disabled button can't produce any messages)

Reply:
@jhrcek I didn't know that `Nothing` disabled the button. But anyway, I don't find the current button's API clean.